### ISSUE TITLE
Module not found: Error: Can't resolve 'os'

### ISSUE THREAD

**Issue by bellizio** [contentful/contentful.js#243]
* **Node Version**: 8.11.1
* **Npm Version**: 5.6.0
* **Platform**: Mac OS 10.13.4
* **Package Version**: 5.1.3

I am using Contentful with Angular and have set up the service in my project following the instructions [here](https://www.contentful.com/developers/docs/javascript/tutorials/using-contentful-in-an-angular-project/). Everything worked perfectly with Angular 5.x, but I get the following error after upgrading to Angular 6:
```
ERROR in ./node_modules/contentful-sdk-core/dist/es-modules/get-user-agent.js
Module not found: Error: Can't resolve 'os' in '/Users/dbellizio/Dev/epri/epri-mcr-poc/node_modules/contentful-sdk-core/dist/es-modules'
```

My project is Angular CLI based, which as of version 6 now uses webpack 4 for the build process. Angular 5 uses webpack 3. So I'm wondering if that has something to do with it. The error occurs when running `ng serve` or any other Angular CLI build command.

For what it's worth, I tried upgrading to version 6.1.0 of Contenful, but the error remains.

**Comment by Khaledgarbaya** [comment]
Hey @bellizio,
Haven't tried the SDK with angular 6 but I will give it try next week and let you know.

Best,
Khaled

**Comment by edyrkaj** [comment]
Same for angular@6.0.0. 
Maybe it is connected with @angular/cli version. It is not working with latest version of @angular/cli

**Comment by Khaledgarbaya** [comment]
Hey @edyrkaj and @bellizio,
would it possible to isolate the problem in a small GitHub project?
This will help a lot to speed up the investigation process.

Best,
Khaled

**Comment by bellizio** [comment]
@Khaledgarbaya sure thing, here you go: https://github.com/bellizio/ng6-contentful

The real-world app I am using Contentful in is much more complex, but the simplified demo app confirms the issue is not a result of other additional dependencies and/or app configuration.

**Comment by Khaledgarbaya** [comment]
Thanks a lot, I will try to debug the issue

**Comment by Khaledgarbaya** [comment]
Hey @bellizio looks like I spotted the issue, looks like the new angular version does not like dynamic require also it is picking the es modules version and tries to build it which actually fails, I think we need top figure out a way to redirect the build system to pick the browser version of the SDK, something like webpack's alias  `contentful: 'contentful/dist/contentful.browser.min.js'`

**Comment by Khaledgarbaya** [comment]
ok 
@bellizio I think solved the issue:

can you change your tsconfig.json to

```js
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "paths":{
      "contentful": ["./node_modules/contentful/dist/contentful.browser.js"]
    },
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2017",
      "dom"
    ]
  }
}
```
and test again

**Comment by bellizio** [comment]
@Khaledgarbaya that worked. I'm not familiar with the "paths" option in tsconfig, though. 

Is this what will be necessary for Angular 6 users moving forward, or is there something that can be done on the contentful side to address the issue?

**Comment by bellizio** [comment]
@Khaledgarbaya sounds good. Thanks for looking into it and for the workaround!

**Comment by briancodes** [comment]
Hi

When I change my `paths` variable as follows, I get TypeScript errors on build:
```
"paths":{
        "contentful": ["../node_modules/contentful/dist/contentful.browser.js"]
      }
```
Errors
```
....node_modules/contentful/dist/contentful.browser"' has no exported member 'Entry'.
```
I also have the  `"allowJs": true` for other imports, and if I set `allowJs` to false, I get errors for my `entryCollection.items.map` code when retrieving collections
```
error TS2339: Property 'items' does not exist on type '{}'.
```

**Comment by Khaledgarbaya** [comment]
Hi @briancodes  this definitely needs a deeper investigation.

**Comment by Khaledgarbaya** [comment]
Hey @briancodes here is a working example with angular6 https://github.com/Khaledgarbaya/contentful-angular6-starter

**Comment by briancodes** [comment]
Hi 

In the example link there doesn't seem to be much Typings support. `Observable<Entry<any>>`,  `books$: Observable<any>`. `map((response: any)` - that is the problem I was having

I'll run the example later and see. Thanks

**Comment by Khaledgarbaya** [comment]
Hey @briancodes we shipped a fix that should solve your problem, **v6.1.1**.

**Comment by bellizio** [comment]
@Khaledgarbaya so just to clarify: the fix in version 6.1.1 means the workaround in tsconfig.json can be removed now?

**Comment by Khaledgarbaya** [comment]
@bellizio yes

**Comment by fhackenb** [comment]
I got this issue for a different library, but solved it by running `npm install --save os`

**Comment by nagkumar** [comment]
I got this error when moving from webpack 4.x to webpack 5.x,

I have included     "os-browserify": "^0.3.0" as devDependencies in pakage.json and also included in webpack.config.js 

```
resolve: {
        fallback: {"os": require.resolve("os-browserify/browser")}
    },

```
